server/src/views/frame.js

```diff
@@ -115,14 +64,14 @@ class Snippet extends React.Component {
     }
 
     return <div className="snippet-container">
-      <a href={'#clip=' + encodeURIComponent(clip.id)}>
+      <a href={'?clip=' + encodeURIComponent(clip.id)}>
         { node }
       </a>
       <p>
         <img ref="commentBubble" className="comment-bubble"
           src={ closed ? this.props.staticLink("img/comment-bubble.png") : this.props.staticLink("img/comment-bubble-open.png") }
           onClick={ this.onClickComment.bind(this) } />
-        <a href={'#clip=' + encodeURIComponent(clip.id)}>
+        <a href={'?clip=' + encodeURIComponent(clip.id)}>
```

Comment:
Then, weird though it is, we can double-render: first render with the server variables, then read location.hash and change the variables and then render again